torch/testing/_internal/common_quantized.py

```diff
@@ -102,6 +102,27 @@ def _calculate_dynamic_per_channel_qparams(X, dtype):
 
     return scale, zero_point
 
+def _snr(x, x_hat):
```

Comment:
nit: can we annotate types on function I/O

Reply:
This is a very good point. The short answer is **maybe**, but I don't see the reason why :)

The long answer is that the files under the `testing/_internal` are not expected to get jitted or to have a mandatory type checking. This means that other than readability the type hints in this folder will serve no purpose.

However, because mot functions under testing can handle multiple argument types, the readability will suffer a lot if type hints are added. For example, here how this function will look like:

```python
from typing import List, Tuple
from torch import Tensor

ListOfTensors = List[Union[List, Tuple, Tensor]]
TupleOfTensors = Tuple[Union[List, Tuple, Tensor]]

def _snr(x: Union[Tensor, ListOfTensors, TupleOfTensors],
         x_hat: Union[Tensor, ListOfTensors, TupleOfTensors]
) -> Union[Tensor, ListOfTensors, TupleOfTensors]:
    ...

# Or, if we are OK with being less descriptive:

def _snr(x: Union[Tensor, List, Tuple],
         x_hat: Union[Tensor, List, Tuple]
) -> Union[Tensor, List, Tuple]:
    ...

```

If you notice, the readability of the code does not improve, and because the function is not expected to get type checked, there are only downsides of annotating it. With that said, if we decide to annotate the rest of the code under the `testing/_internal`, I think we can start now

Reply:
> However, because mot functions under testing can handle multiple argument types, the readability will suffer a lot if type hints are added. For example, here how this function will look like:

The answer might vary per person, but having types significantly improves readability for me personally, even if it might decrease it for you.  Are there any compromises we can go for here?

Reply:
Wait, I thought I already added args description in the code to avoid the type annotations. Can you, please, refresh?

Reply:
```
ListOfTensors = List[Union[List, Tuple, Tensor]]
TupleOfTensors = Tuple[Union[List, Tuple, Tensor]]

def _snr(x: Union[Tensor, ListOfTensors, TupleOfTensors],
         x_hat: Union[Tensor, ListOfTensors, TupleOfTensors]
) -> Union[Tensor, ListOfTensors, TupleOfTensors]:
```

if you're up for doing this, that would be awesome.  I think it's much more readable than not having it.